package/network/services/hostapd/files/hostapd.sh

```diff
@@ -484,7 +485,15 @@ hostapd_set_bss_options() {
 		;;
 		psk|sae|psk-sae)
 			json_get_vars key wpa_psk_file
-			if [ ${#key} -eq 64 ]; then
+			if [ "$auth_type" = "psk" ] && [ "$ppsk" -ne 0 ] ; then
+				json_get_vars auth_server auth_secret auth_port
+				set_default auth_port 1812
+				append bss_conf "auth_server_addr=$auth_server" "$N"
+				append bss_conf "auth_server_port=$auth_port" "$N"
+				append bss_conf "auth_server_shared_secret=$auth_secret" "$N"
+				append bss_conf "macaddr_acl=2" "$N"
+				append bss_conf "wpa_psk_radius=2" "$N"
```

Comment:
As eluded to in https://github.com/openwrt/openwrt/pull/3509#issuecomment-749548980, this PR is a bit behind the current stable version. After manually applying this diff to hostapd.sh from v19.07.6, I also had to add the following line here in order to make dynamic VLANs work

```
vlan_possible=1
```

Afterwards, this works beautifully!

Reply:
Yes, if you are on 19.07 or older you will have to do vlan_possible=1, master has it already at the end of the case statement and it is not required. Thank you @bchavet for pointing it out for other people that may be interested in using it outside of master :D